arch/risc-v/src/qemu-rv/qemu_rv_start.c

```diff
@@ -144,9 +148,25 @@ void qemu_rv_start(int mhartid)
 
 #ifdef CONFIG_BUILD_KERNEL
 
+static bool g_bss_cleared = false;
+
 void qemu_rv_start_s(int mhartid)
 {
-  qemu_rv_clear_bss();
+  if (0 == mhartid)
+    {
+      qemu_rv_clear_bss();
+      g_bss_cleared = true;
+      __DMB();
+    }
+#ifdef CONFIG_SMP
+  else
+    {
+      while (!g_bss_cleared)
```

Comment:
If the secondary core boot too fast(before primary core clear .bss section), g_bss_cleared is a random number.

Reply:
Here is my summary:

1. It's enough to check hartid to avoid the double initialization
2. To avoid the secondary cores run before the primary core finish the basic initialization, we need either wait a special SGI sent from primary code or a variable which is clear by hardware(NOT software) when power on but set by primary code.

Reply:
OK, let me consider this PR again.